Userland/Libraries/LibC/unistd.cpp

```diff
@@ -975,8 +975,41 @@ int unveil(char const* path, char const* permissions)
 // https://pubs.opengroup.org/onlinepubs/9699919799/functions/getpass.html
 char* getpass(char const* prompt)
 {
-    dbgln("FIXME: getpass('{}')", prompt);
-    TODO();
+    int tty = open("/dev/tty", O_RDWR | O_NOCTTY | O_CLOEXEC);
+
+    if (tty < 0)
+        return nullptr;
+
+    struct termios backup, noecho;
+    tcgetattr(tty, &backup);
+
+    noecho = backup;
+    noecho.c_lflag &= ~(ECHO|ISIG|INLCR|IGNCR);
+    noecho.c_lflag |= ICANON | ICRNL;
+
+    tcsetattr(tty, TCSAFLUSH, &noecho);
+    tcdrain(tty);
+
+    write(tty, prompt, strlen(prompt));
+
+    static char password[PASS_MAX];
+    ssize_t chars_read = read(tty, password, sizeof(password));
```

Comment:
should password be zero filled before calling read, to clear prior content (security concern)?

Reply:
More importantly, why is this `static` at all?

Reply:
Probably static because that's what Dr. Posix says to do. 🤔

>        The function getpass() returns a pointer to a static buffer
>       containing (the first PASS_MAX bytes of) the password without the
>       trailing newline, terminated by a null byte ('\0').  This buffer
>       may be overwritten by a following call.  On error, the terminal
>       state is restored, errno is set to indicate the error, and NULL
>       is returned.

Reply:
> should password be zero filled before calling read, to clear prior content (security concern)?

From man page: "The calling process should zero the password as soon as possible to avoid leaving the cleartext password visible in the process's address space."

It seems that would be out of scope for this function.

> More importantly, why is this `static` at all?

In order for it to be stored in persistant memory. A local variable on the stack would be destroyed after the function returns.